packages/plugin/src/VisualizationPlugin.js

```diff
@@ -65,23 +62,31 @@ export const VisualizationPlugin = ({
         }
 
         return result
-    }, [d2, filters, forDashboard, onResponsesReceived, visualization])
+    }, [engine, filters, forDashboard, onResponsesReceived, visualization])
 
     const doFetchLegendSets = useCallback(
         async legendSetIds => {
             if (!legendSetIds.length) {
                 return []
             }
 
-            const response = await apiFetchLegendSets(engine, legendSetIds)
+            const legendSets = await apiFetchLegendSets(engine, legendSetIds)
 
-            if (response && response.legendSets) {
-                return response.legendSets.legendSets
-            }
+            return legendSets
         },
         [engine]
     )
 
+    useEffect(() => {
+        const doFetchOuLevels = async () => {
+            const ouLevels = await apiFetchOrganisationUnitLevels(engine)
+
+            setOuLevels(ouLevels)
+        }
+
+        doFetchOuLevels()
+    }, [engine])
```

Comment:
Does the engine change? Does it have to be in the array, or does it work as expected with an empty array? (Not sure if this would be preferred over the current, actually)

Reply:
It shouldn't change, but having it avoids the linting errors.